### ISSUE TITLE
build config error, missing aspectpath entry/incorrect classpath

### ISSUE THREAD

**Issue by jeffcharles** [JakeWharton/hugo#121]
When I try to add Hugo to my build, I get the following during the `:app:compileDebugJavaWithJavac` task:

```
build config error: skipping missing, empty or corrupt aspectpath entry: /Users/jeff/projects/VisitorDetector/app/build/tmp/kotlin-classes/debug
incorrect classpath: /Users/jeff/projects/VisitorDetector/app/build/tmp/kotlin-classes/debug
advice defined in hugo.weaving.internal.Hugo has not been applied [Xlint:adviceDidNotMatch]
```

and I don't get any of the logging output from Hugo.

Any idea as to what could be going wrong?

The code I'm trying to build with `./gradlew build` is at https://github.com/jeffcharles/visitor-detector/tree/add-hugo (specifically the `add-hugo` branch).

**Comment by JakeWharton** [comment]
Pretty sure it's as simple as the fact that the plugin just can't handle
the Kotlin plugin. Right now it hooks into the JavaCompile tasks so Kotlin
sources will never be processed (hence the did not match warning). Changing
this to a Gradle transform instead of a hacky plugin that hooks into tasks
will fix this.

**Comment by jeffcharles** [comment]
Fair enough. Might be worth calling out that Hugo doesn't support non-Java Android languages for now in the README.

**Comment by okmanideep** [comment]
@JakeWharton Any intention of 

> Changing this to a Gradle transform instead of a hacky plugin that hooks into tasks

? (is kotlin support in the pipeline?)

**Comment by rfgamaral** [comment]
I'm also getting this and I'm not using Kotlin at all:

> :app:compileDebugJavaWithJavac
> advice defined in hugo.weaving.internal.Hugo has not been applied [Xlint:adviceDidNotMatch](...)
> :app:assembleDebug

**Comment by hiteshandroid** [comment]
> Pretty sure it's as simple as the fact that the plugin just can't handle
> the Kotlin plugin. Right now it hooks into the JavaCompile tasks so Kotlin
> sources will never be processed (hence the did not match warning). Changing
> this to a Gradle transform instead of a hacky plugin that hooks into tasks
> will fix this.
> 
> On Fri, Jan 29, 2016 at 10:18 PM Jeffrey Charles [notifications@github.com](mailto:notifications@github.com)
> wrote:
> 
> > When I try to add Hugo to my build, I get the following during the
> > :app:compileDebugJavaWithJavac task:
> > build config error: skipping missing, empty or corrupt aspectpath entry: /Users/jeff/projects/VisitorDetector/app/build/tmp/kotlin-classes/debug
> > incorrect classpath: /Users/jeff/projects/VisitorDetector/app/build/tmp/kotlin-classes/debug
> > advice defined in hugo.weaving.internal.Hugo has not been applied [Xlint:adviceDidNotMatch]
> > and I don't get any of the logging output from Hugo.
> > Any idea as to what could be going wrong?
> > The code I'm trying to build with ./gradlew build is at
> > https://github.com/jeffcharles/visitor-detector/tree/add-hugo
> > (specifically the add-hugo branch).
> > —
> > Reply to this email directly or view it on GitHub
> > #121.

@JakeWharton. We are currently migrating our android projects to Kotlin. Is the support for Kotlin to be released soon ?
